tools/integration_tests/util/client/storage_client.go

```
@@ -123,6 +128,17 @@ func ReadChunkFromGCS(ctx context.Context, client *storage.Client, object string
 	return string(content), nil
 }
 
+func NewWriterExt(ctx context.Context, o *storage.ObjectHandle, client *storage.Client) *storage.Writer {
```

IMO we should log an error when we get a zonal bucket here when the zonal flag is not set. Will help us in catching incorrect usage.

---

P.S. I was looking for something similar to what you've done at https://github.com/GoogleCloudPlatform/gcsfuse/pull/2979/files#diff-6f7c497fa7d6409c46108caba011760f35575256bc2e3169948a2d686f780c5d.
But instead of panic, we can log an error. Let me know what you think.

Also, can we think of a better name for this function? How about NewConfiguredWriter?

---

> I was looking for something similar to what you've done at

yes, that is what I meant by my next PR.

> But instead of panic, we can log an error

The reason I kept a panic instead of the returning of returning error, was to keep the same signature as that of NewWriter, which also doesn't return error. Anyway, I have changed it to returning error now.